host/libhackrf/src/hackrf.h

```diff
@@ -43,88 +43,288 @@ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSI
 #else /* _WIN32 not defined. */
 
 	/* Define with no value on non-Windows OSes. */
+	/**
+	 * __declspec(dllimport / dllexport) on windows, unused on other platforms
+	 */
 	#define ADDAPI
+	/**
+	 * __cdecl calling convention on windows, unused on other platforms
+	 */
 	#define ADDCALL
 
 #endif
 
+/**
+ * Number of samples per tuning when sweeping
+ */
 #define SAMPLES_PER_BLOCK 8192
+
+/**
+ * Number of bytes per tuning for sweeping
+ */
 #define BYTES_PER_BLOCK   16384
+
+/**
+ * Maximum number of sweep ranges to be specified for @ref hackrf_init_sweep
+ */
 #define MAX_SWEEP_RANGES  10
 
+/**
+ * Invalid operacake add-on board address, placeholder in @ref hackrf_get_operacake_boards
+ */
 #define HACKRF_OPERACAKE_ADDRESS_INVALID 0xFF
+
+/**
+ * Maximum number of connected operacake add-on boards
+ */
 #define HACKRF_OPERACAKE_MAX_BOARDS      8
+
+/**
+ * Maximum number of specifiable dwell times for operacake add-on boards
+ */
 #define HACKRF_OPERACAKE_MAX_DWELL_TIMES 16
+
+/**
+ * Maximum number of specifiable frequency ranges for operacake add-on boards
+ */
 #define HACKRF_OPERACAKE_MAX_FREQ_RANGES 8
 
+/**
+ * error enum, returned by many libhackrf functions
+ * 
+ * Many functions that are specified to return INT are ctually returning this enum
+ */
 enum hackrf_error {
+	/**
+	 * no error happened
+	 */
 	HACKRF_SUCCESS = 0,
+	/**
+	 * TRUE value, only used by hackrf_is_streaming
```

Comment:
Since this breaks "conventional" error handling (`!result` or `result != HACKRF_SUCCESS`) I do think this case requires some special attention, mostly at the documentation of the functions that return this specific value, but a mention of this odd-ish behavior is worth mentioning here in my opinion. I agree that a list of functions is not future-proof though, so I guess I'll remove that.